ipn/ipnlocal/local.go

```diff
@@ -268,6 +268,7 @@ type LocalBackend struct {
 	keyExpired       bool
 	authURL          string    // non-empty if not Running
 	authURLTime      time.Time // when the authURL was received from the control server
+	interact         bool      // indicates whether a user requested interactive login
```

Comment:
We worked so hard to remove this bool :)

How much have you tested the CLI with this change and "tailscale up"?

Reply:
Ultimately, I think we want to make `(*LocalBackend).StartLoginInteractive` synchronous. Regardless, we need to store the interactive login state somewhere. Currently, this is just a `bool`, but in a follow-up PR, I'm changing it to an `ipnauth.Actor`. With a synchronous login, we'll likely need to store a `context.CancelFunc` so that it can be canceled in scenarios like switching profiles.

I'm not perfectly sure where our desire to remove that bool comes from. But it appeared to me that for quite some time prior to https://github.com/tailscale/tailscale/pull/12096 we just used it wrongly. We never sent the `authURL` to the client unless it initiated an interactive login, which was wrong and caused issues with `tailscale up` and similar scenarios where the control sends us an `authURL` due to an expired key or just because the node has never been authenticated before.

I'm finding it hard to quantify how much I tested `tailscale up`, but I did test it. I'm happy to check any additional scenarios you can think of. But I also believe that the proposed change is straightforward and easy to reason about: we keep sending `authURL` in the same scenarios as in https://github.com/tailscale/tailscale/pull/12096 with the `interact` flag removed, but we also send it to the client if an interactive login has been requested.